pkg/admission/validator/shoot.go

```diff
@@ -164,8 +168,7 @@ func validateAuditConfigMap(decoder runtime.Decoder, configMap *corev1.ConfigMap
 	}
 
 	auditdConfig := &rsyslog.Auditd{}
-
-	_, _, err := decoder.Decode([]byte(auditdConfigString), nil, auditdConfig)
+	err := runtime.DecodeInto(decoder, []byte(auditdConfigString), auditdConfig)
```

Comment:
Sorry, just noticed this on a second look - could you also use one line for the error checking in all occurrences? 
```suggestion
	if err := runtime.DecodeInto(decoder, []byte(auditdConfigString), auditdConfig); err != nil {
```